src/containers/Auth/SignIn/SignIn.js

```diff
@@ -0,0 +1,167 @@
+import React, { Component } from "react";
+import { Form, Button, Grid, Header } from "semantic-ui-react";
+import { SignUpLink } from "../SignUp/SignUp";
+import { Link, withRouter } from "react-router-dom";
+// import * as ROUTES from '../../constants/routes';
+import { loginUser } from "../../../redux/actions/userAction";
+import { connect } from "react-redux";
+import { Redirect } from "react-router-dom";
+import formHoc from "../../../hoc/formHoc";
+
+const INITIAL_STATE = {
+  email: "",
+  password: "",
+  error: null,
+};
+
+export class SignInFormBase extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = { ...INITIAL_STATE };
+  }
+  emailValidation() {
+    const regex =
```

Comment:
you can separate out validation.js file, where you commonly write the validation rules and use as res-usable file for all type of forms.